cmake/fetch_boost.cmake

```diff
@@ -0,0 +1,24 @@
+include(${PROJECT_SOURCE_DIR}/cmake/CPM.cmake)
+
+
+function(fetch_boost_library BOOST_PROJECT_NAME)
+    CPMAddPackage(
+        NAME boost_${BOOST_PROJECT_NAME}
+        GITHUB_REPOSITORY boostorg/${BOOST_PROJECT_NAME}
+        GIT_TAG boost-1.85.0
+    )
+
+  get_target_property(BOOST_INTERFACE_LINK_LIBRARIES
+                      boost_${BOOST_PROJECT_NAME} INTERFACE_LINK_LIBRARIES)
+
+  if(NOT BOOST_INTERFACE_LINK_LIBRARIES STREQUAL
```

Comment:
Is this a recursive call which in effect ensures that we only fetch each boost library that we need once because we should only be touching each one we're using + their upstream dependencies in boost once through checking if they're targets?

Reply:
Yep, so we can just specify the top boost dependencies and this function pull all their children.